src/main/java/store/controller/StoreController.java

```diff
@@ -0,0 +1,99 @@
+package store.controller;
+
+import static store.util.InputProcessor.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import store.service.PurchaseService;
+import store.service.StoreService;
+import store.util.InputFormatter;
+import store.util.Reader;
+import store.view.InputView;
+import store.view.OutputView;
+
+public class StoreController {
+
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final StoreService storeService;
+    private PurchaseService purchaseService;
+
+    public StoreController() {
+        this.inputView = new InputView();
+        this.outputView = new OutputView();
+        this.storeService = new StoreService();
+    }
+
+    public void run() {
+        setStore();
+
+        while (true) {
+            this.processPurchase();
+            if (!continueUntilNormalInput(this::processContinueInput)) {
+                break;
+            }
+        }
+    }
+
+    private void setStore() {
+        storeService.set(Reader.readProducts(), Reader.readPromotions());
+    }
+
+    private void processPurchase() {
+        this.purchaseService = new PurchaseService();
+        printStoreInformation();
+        processInput();
+        purchaseService.applyPurchases();
+        outputView.printReceipt(purchaseService.getPurchasesContent(), purchaseService.getGifts(), purchaseService.getAmounts());
+    }
+
+    private void printStoreInformation() {
+        outputView.printWelcomeMessage();
+        outputView.printProductsInformation(storeService.getProductsInformation());
+    }
+
+    private void processInput() {
+        continueUntilNormalInput(this::processPurchaseInput);
+        continueUntilNormalInput(this::processAddQuantityInput);
+        continueUntilNormalInput(this::processPurchaseConfirmInput);
+        continueUntilNormalInput(this::processMembershipConfirmInput);
```

Comment:
```suggestion
        List<Runnerable> inputProcesses = List.of(
                this::processPurchaseInput,
                this::processAddQuantityInput,
                this::processPurchaseConfirmInput,
                this::processMembershipConfirmInput
            );
            
            inputProcesses.forEach(InputProcessor::continueUntilNormalInput);
```

이렇게 고치면 새로운 입력 처리가 생겼을 때 대처하기 쉽지 않을까요?

Reply:
오.. 이렇게 `List`로 관리할 생각은 아예 못했는데, 말씀해주신 것처럼 확장성의 측면에서 더 좋은 방법인 것 같습니다!